File: lua/fizzbuzz.lua
local max_number = tonumber(arg[1]) -- Get the maximum number from the command line argument

if not max_number then
    print("Please provide a number as an argument.") -- Print an error message if no number is provided
    return
end

for i = 1, max_number do
    if i % 3 == 0 and i % 5 == 0 then -- Check if the number is divisible by both 3 and 5
        print("FizzBuzz") -- Print "FizzBuzz" if it is
    elseif i % 3 == 0 then -- Check if the number is divisible by 3
        print("Fizz") -- Print "Fizz" if it is
    elseif i % 5 == 0 then -- Check if the number is divisible by 5
        print("Buzz") -- Print "Buzz" if it is
    else
        print(i) -- Print the number itself if none of the above conditions are met
    end
end

File: java/fizzbuzz.java
public class fizzbuzz {
    public static void main(String[] args) {
        // Check if a command-line argument is provided
        if (args.length == 0) {
            System.out.println("Please provide a number as a command-line argument.");
            return;
        }

        // Parse the maximum number from the command-line argument
        int maxNumber = Integer.parseInt(args[0]);

        // Iterate from 1 to the maximum number
        for (int i = 1; i <= maxNumber; i++) {
            // Check if the current number is divisible by both 3 and 5
            if (i % 3 == 0 && i % 5 == 0) {
                System.out.println("FizzBuzz");
            }
            // Check if the current number is divisible by 3
            else if (i % 3 == 0) {
                System.out.println("Fizz");
            }
            // Check if the current number is divisible by 5
            else if (i % 5 == 0) {
                System.out.println("Buzz");
            }
            // If none of the above conditions are met, print the number itself
            else {
                System.out.println(i);
            }
        }
    }
}